File: frontend/services/api/backgrounds.ts
import { httpClient } from './base';
import { ApiResponse } from '../../types';

export interface BackgroundImage {
  filename: string;
  display_name: string;
}

export interface BackgroundListData {
  backgrounds: BackgroundImage[];
}

export interface BackgroundUploadData {
  filename: string;
  display_name: string;
}

export const backgroundsApi = {
  getBackgrounds: async (): Promise<ApiResponse<BackgroundListData>> => {
    return httpClient.get<BackgroundListData>('/backgrounds');
  },

  uploadBackground: async (file: File): Promise<ApiResponse<BackgroundUploadData>> => {
    const formData = new FormData();
    formData.append('file', file);
    return httpClient.post<BackgroundUploadData>('/backgrounds/upload', formData);
  },
};

File: api/routes/backgrounds.py
from fastapi import APIRouter, File, UploadFile

from api.schemas import ResponseModel
from core.services.background_service import list_backgrounds, save_uploaded_background


router = APIRouter()


@router.get("/backgrounds", response_model=ResponseModel)
async def get_backgrounds():
    backgrounds = [item.to_dict() for item in list_backgrounds()]
    return {
        "code": 200,
        "message": "success",
        "data": {"backgrounds": backgrounds},
    }


@router.post("/backgrounds/upload", response_model=ResponseModel)
async def upload_background(file: UploadFile = File(...)):
    background = save_uploaded_background(file)
    return {
        "code": 200,
        "message": "上传成功",
        "data": {
            "filename": background.filename,
            "display_name": background.display_name,
        },
    }
